### DISCUSSION TITLE
Interpolation with Pygment highlighted text / Custom lexer

### DISCUSSION THREAD

**Discussion by yodahuang** [Textualize/rich#985] [Q&A]
Is there any way for `rich` to use highlighted Pygment text? For example:
```python
from pygments.formatters import BBCodeFormatter
from pygments.lexers.shell import BashLexer
from pygments import highlight

from rich.console import Console

console = Console()
console.print(highlight("cd /tmp", BashLexer(), BBCodeFormatter(style="emacs")))
```
You can see the highlighting is off since rich does not understand BBCode, and there doesn't seem to be a formatter that `rich` can understand. Reading the source code, `rich` is not using any formatter provided by `pygments` anyway.

Why do I need that? I need a custom lexer, e.g.
```python
class BetterBashLexer(BashLexer):
    tokens = {
        'basic': [(r'\b(bazel build|bazel run|bazel test|bb|br|bt)(?=[\s)`])', token.Name.Function)
                  ],
        'data': [(r'(?<=:)\w+', token.Name.Variable)]  # Bazel target
    }
```
but `rich` does not have the option to pass my custom lexer in for `Syntax`, so I need to either use `pygments` to render my text, or need `rich` to support custom lexers.

**Comment by willmcgugan** [comment]
Rich doesn't support passing in a lexer at the moment, but it could. Probably the "lexer" argument could accept a string or a Lexer instance. File a feature request if you would like that, or feel free to submit a PR if you would like to tackle it yourself.
